machine/ingrasys/ingrasys_s8900_54xc/kernel/0001-driver-i2c-bus-intel-ismt-set-default-bus-speed.patch

```diff
@@ -0,0 +1,28 @@
+From 87a28878d49a9b4a3fc8c4d882b8786790f3044d Mon Sep 17 00:00:00 2001
+From: feng <feng@sonic>
+Date: Fri, 18 Aug 2017 11:04:07 +0800
+Subject: [PATCH] driver i2c bus intel ismt set default bus speed
+
+---
+ drivers/i2c/busses/i2c-ismt.c | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/drivers/i2c/busses/i2c-ismt.c b/drivers/i2c/busses/i2c-ismt.c
+index 09a3e03..88ceaa3 100644
+--- a/drivers/i2c/busses/i2c-ismt.c
++++ b/drivers/i2c/busses/i2c-ismt.c
+@@ -192,9 +192,9 @@ static DEFINE_PCI_DEVICE_TABLE(ismt_ids) = {
+ MODULE_DEVICE_TABLE(pci, ismt_ids);
+ 
+ /* Bus speed control bits for slow debuggers - refer to the docs for usage */
+-static unsigned int bus_speed;
++static unsigned int bus_speed = 100;
+ module_param(bus_speed, uint, S_IRUGO);
+-MODULE_PARM_DESC(bus_speed, "Bus Speed in kHz (0 = BIOS default)");
++MODULE_PARM_DESC(bus_speed, "Bus Speed in kHz (100 by default)");
```

Comment:
instead of patching the kernel, you can pass kernel command line parameters to set the i2c-ismt bus speed.

Add this to the your machine.make:
```
EXTRA_CMDLINE_LINUX = i2c-ismt.bus_speed=100
```

For example see commit 5740ba.